### DISCUSSION TITLE
Retrieve a record in a database in notion page with its id

### DISCUSSION THREAD

**Discussion by hpallantla** [ramnes/notion-sdk-py#196] [Q&A] [answered]
I had a table in a notion page, and I am pulling the data from that notion page using a Python client , but now I want to retrieve a record with its id. Is there a way to do the same using a Python client?

**Comment by ramnes** [comment, marked as answer]
Unless I did not understand what you're asking, it should be just the same as any other page: `notion.pages.retrieve(page_id=page_id)`. :)

**Reply by hpallantla** [reply]
Hi @ramnes , Thanks for the immediate response.

Actually i am using the below code snippet to retrieve all the data from the database in notion.

results = notion.databases.query(
    database_id=databaseId)

but i want to retrieve single record using id of the record, is there a way to do this.

**Reply by ramnes** [reply]
Yes, like I said just above, you can get a single page with `notion.pages.retrieve`.

For example:

```python
response = notion.databases.query(database_id=database_id)
page = notion.pages.retrieve(page_id=response["results"][0]["id"])
```

This will get you this page alone (a "record" in a Notion database is a page like any other page in Notion).

Please be more explicit and precise if that's not what you're looking for.

**Comment by hpallantla** [comment]
Thanks a lot @ramnes , it worked !!
